### ISSUE TITLE
Update to JDK8 and add java.nio.Path support

### ISSUE THREAD

**Issue by andresluuk** [zeroturnaround/zt-zip#105]
In one of our applications we are updating Files to Path-s, but this util still accepts only File as inputs.
We should add a Parallel API for Path support. 
Also maybe deprecate the File methods? I would not remove them, for backwards compatibility.
Because Path is introduced in JDK7 we must upgrade the JDK also. I would upgrade to JDK8 because 7 isn't used much and that would allow us to use JDK8 features.

**Comment by toomasr** [comment]
How about adding the Path equivalent methods? As File isn't deprecated in the JDK we should not either. Bumping the minimum version to Java 8 makes sense!

**Comment by eku** [comment]
Will this break compatibility with Android?

**Comment by eku** [comment]
@toomasr did you check with your local Android gurus?

**Comment by toomasr** [comment]
Yup and the feedback was to just maintain the latest good version for Android. I think this is good enough. Once the jdk8 branch lands we'll figure out the versioning. Probably we'll release the Path approach as 2.0 and make sure that 1.x is Android compatible.
